### PR TITLE
feat: add domain blocklist feature

### PR DESCRIPTION
**Pull request by scmmishra**

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@ en:
       invalid: Invalid events
     signup:
       disposable_email: We do not allow disposable emails
+      blocked_domain: This domain is not allowed, if you think this is a mistake please contact support
```
Comment (pranavrajs) [inline config/locales/en.yml:46]:
```suggestion
      blocked_domain: This domain is not allowed. If you believe this is a mistake, please contact support.
```

File: app/builders/account_builder.rb [hunk 2]
```diff
@@ -79,4 +82,20 @@ def create_user
     @user.confirm if @confirmed
     @user.save!
   end
+
+  def domain_blocked?(email)
+    domain = email.split('@').last
+    # loop over blocked_domains and do a regex match
```
Comment (pranavrajs) [inline app/builders/account_builder.rb:88]: Do we need this comment?

**Review by pranavrajs — APPROVED** [review]
Minor comments, otherwise looks fine.

File: app/builders/account_builder.rb (hunk 2 above)
Comment (scmmishra) [reply app/builders/account_builder.rb:88]: Oops, added it for copilot :P 
removing it

**Review by vishnu-narayanan — APPROVED** [review]
Can we also add a spec?

**Comment by scmmishra** [comment]
> Can we also add a spec?

On it

**Pull request merged**